sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts

```diff
@@ -82,8 +187,55 @@ export class TableCheckpointStore implements CheckpointStore {
    *  - `tracingOptions`: Options for configuring tracing.
    * @returns A promise that resolves when the checkpoint has been updated.
    */
-  async updateCheckpoint(/* checkpoint: Checkpoint */): Promise<void> {
-    console.log("no checkpoint to update");
-    throw new Error("not implemented");
+  async updateCheckpoint(checkpoint: Checkpoint) {
+    let PARTITIONKEY =
+      checkpoint.eventHubName +
+      " " +
+      checkpoint.fullyQualifiedNamespace +
+      " " +
+      checkpoint.consumerGroup +
+      " " +
+      "Checkpoint";
+
+    const entity1: customCheckpoint = {
+      partitionKey: PARTITIONKEY,
+      rowKey: checkpoint.partitionId,
+      consumerGroup: checkpoint.consumerGroup,
+      fullyQualifiedNamespace: checkpoint.fullyQualifiedNamespace,
+      eventHubName: checkpoint.eventHubName,
+      sequenceNumber: checkpoint.sequenceNumber,
+      offset: checkpoint.offset,
+      partitionId: checkpoint.partitionId
+    };
+
+    let entitiesIter = this._tableClient.listEntities<Checkpoint>({
+      queryOptions: { filter: odata`PartitionKey eq ${PARTITIONKEY}` }
+    });
+    let i = 0;
+    for await (const ent of entitiesIter) {
+      ent.offset;
+      i++;
+    }
```

Comment:
Why are you calling `listEntities` and iterating them here?

Reply:
i was doing a client-side check again. but this time i wanted to know if there entities with a matching partitionkey. If there are , the iterator i will increment , if not it will remain at 0. Then it will know that such entities do not exist, so the checkpoint is upserted.

Reply:
But you should always `upsert` the checkpoint, right? When would you not upsert it?

Reply:
i will do an update if the entity exists , if it doesnt i will upsert. From the tables documentation, upsert inserts a new entity, and update updates some fields of an existing entity

Reply:
it is not necessary anyway since it is all client-side checks.

Reply:
Gotcha. `upsert` is actually pretty neat in that it will either insert a new entity, or update an existing entity (it's a mash-up of 'Update' and 'Insert'.) The main reason to use `update` instead of `upsert` in some cases is if you need to pass the etag (which isn't required for checkpoints)